invenio_communities/communities/schema.py

```diff
@@ -76,6 +76,13 @@ def is_not_uuid(value):
         pass
 
 
+def is_not_reserved(value):
+    """Make sure value is not a reserved keyword."""
+    return validate.NoneOf(
+        iterable=["new"], error=("The ID cannot be a reserved keyword.")
```

Comment:
I wouldn't go about changing URLs, since these are often externally referenced in project pages or reports.

As mentioned in https://github.com/inveniosoftware/invenio-communities/pull/1210#discussion_r1774658800, having this being dynamically computed (and then cached) from the registered Flask routes would be the most future-proof.

Reply:
this won't work if the community with slug `featured` is first created and then we want to create a route `communities/featured`. I think we should change the root path

Reply:
I think this approach of changing the URL structure is very radical and should be well considered (as URLs should last [2 years, 20 years, even 200 years](https://www.w3.org/Provider/Style/URI)). 

For a user is it not confusing that to create a record you go to `/uploads/new` but `/new/communities/create`? Or is the idea that we going to change all URLs so that they are verb first?

I don't think it's unreasonable to introduce a controlled vocabulary of names that communities aren't allowed, as Github and all social medias do this and so people are understanding. Perhaps when we create a new release of Invenio, in the rare case when we add a reserved keyword, it could be part of the migration instructions to tell maintainers to ask to change the name of the community or disable the feature.

The only reason I'm against the verb first URLs is that it is very easy to create URLs with subpaths that 404. Users of library systems will expect to be able to modify the URL to where they want to go and it would be confusing if `/featured` or '/new/communities` 404s. If we're willing to do the extra work to make sure the URLs are consistent and don't 404, then I think switching to verb first makes sense